### PR TITLE
fix editor dismissing upon background in iPadOS

### PR DESCRIPTION
**Pull request by tonisevener**
& fix UITableViewAlertForLayoutOutsideViewHierarchy console warning

https://phabricator.wikimedia.org/T229494

I couldn't repro the freezing anymore but editor would dismiss if you had table of contents open in article > go to editor > background for 5-10 seconds. This is because oddly only in iPadOS willTransitionToTraitCollection in WMFArticleViewController is called twice upon background - the size classes switch to compact then switch back to what they were before. We try presenting table of contents again in this situation assuming the user is split screening which seems to automatically dismiss the editor. Fix is to make sure WMFArticleViewController isn't presenting anything like editor before attempting to present the table of contents.

### CODE REVIEW

File: Wikipedia/Code/WMFTableOfContentsViewController.swift [hunk 1]
```diff
@@ -224,8 +224,8 @@ open class WMFTableOfContentsViewController: UIViewController, UITableViewDelega
         tableOfContentsFunnel.logOpen()
     }
     
-    open override func viewDidDisappear(_ animated: Bool) {
-        super.viewDidDisappear(animated)
+    open override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
         deselectAllRows()
```
Comment (nambatee) [inline Wikipedia/Code/WMFTableOfContentsViewController.swift]: Is it necessary to deselect all rows at all? If I background the app and come back, I'm scrolled to the right section but the selected section in the TOC is off

Comment (montehurd) [reply Wikipedia/Code/WMFTableOfContentsViewController.swift]: hmm.. wonder if my tweak to [TOC fragment logic](https://github.com/wikimedia/wikipedia-ios/pull/3230) affects this? will check

Comment (montehurd) [reply Wikipedia/Code/WMFTableOfContentsViewController.swift]: i may know a fix for the TOC issue @nambatee noticed... checking...

Comment (tonisevener) [reply Wikipedia/Code/WMFTableOfContentsViewController.swift]: Sounds good, this was just to get rid of that console error I was seeing, didn't affect the editor presentation dismissal upon backgrounding.

Comment (montehurd) [reply Wikipedia/Code/WMFTableOfContentsViewController.swift]: TOC selection reproducible on 6.3. Filing separate ticket with repro steps.

**Pull request merged**